test/test_reads_at_locus.py

```diff
@@ -180,3 +185,32 @@ def test_reads_at_locus_substitution_shorter():
         base0_read_position_before_variant=0,
         base0_read_position_after_variant=2)
     assert_equal_fields(read, expected)
+
+def test_reads_at_locus_dataframe():
+    sam_all_variants = load_bam("data/b16.f10/b16.combined.bam")
+
+    n_reads_expected = 0
+
+    sam_path_single_variant = data_path(
+        "data/b16.f10/b16.f10.127a.aldh1b1.chr4.45802539.refG.altC.sam")
+    with open(sam_path_single_variant) as f:
+        for line in f:
+            if line.startswith("HWI"):
+                n_reads_expected += 1
+    # we know from inspecting the file that *one* of the reads overlapping this
+    # variant has a CIGAR string of N at the location before and thus we'll
```

Comment:
It shouldn't be! But the current logic drops that read, which is a bug to
be fixed in the next PR.
On Apr 22, 2016 2:16 PM, "Isaac Hodes" <notifications@github.com> wrote:

> In test/test_reads_at_locus.py
> <https://github.com/hammerlab/isovar/pull/15#discussion_r60780994>:
>
> > @@ -180,3 +185,32 @@ def test_reads_at_locus_substitution_shorter():
> >          base0_read_position_before_variant=0,
> >          base0_read_position_after_variant=2)
> >      assert_equal_fields(read, expected)
> > +
> > +def test_reads_at_locus_dataframe():
> > +    sam_all_variants = load_bam("data/b16.f10/b16.combined.bam")
> > +
> > +    n_reads_expected = 0
> > +
> > +    sam_path_single_variant = data_path(
> > +        "data/b16.f10/b16.f10.127a.aldh1b1.chr4.45802539.refG.altC.sam")
> > +    with open(sam_path_single_variant) as f:
> > +        for line in f:
> > +            if line.startswith("HWI"):
> > +                n_reads_expected += 1
> > +    # we know from inspecting the file that *one* of the reads overlapping this
> > +    # variant has a CIGAR string of N at the location before and thus we'll
>
> And why would this would be excluded from "reads_at_locus_dataframe"?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hammerlab/isovar/pull/15/files/e7077b31c46d6a248d8f12099f6d23c392cadc41#r60780994>
>